bondcpp/src/bond.cpp

```diff
@@ -538,16 +539,16 @@ void Bond::publishStatus(bool active)
   msg->id = id_;
   msg->instance_id = instance_id_;
   msg->active = active;
-  msg->heartbeat_timeout = heartbeat_timeout_;
-  msg->heartbeat_period = heartbeat_period_;
+  msg->heartbeat_timeout = heartbeat_timeout_.seconds();
+  msg->heartbeat_period = heartbeat_period_.seconds();
   pub_->publish(std::move(msg));
 }
 
 void Bond::flushPendingCallbacks()
 {
-  std::vector<std::function<void(void)>> callbacks;
+  std::vector<EventCallback> callbacks;
   {
-    std::unique_lock<std::mutex> lock(mutex_);
+    std::unique_lock<std::mutex> lock(callbacks_mutex_);
```

Comment:
I feel like the `callbacks_mutex` isn't taken in enough places to effectively guard `pending_callbacks_`.  In particular, it is not being taken while adding to the `pending_callbacks_` vector in `BondSM::Connected` and `BondSM::Death`.

Reply:
Increased places I am locking.